Package & Tests Broken: Errors internal to this package must be fixed to function

After forking this package, it became apparent that there are errors being thrown within this package causing it to fail to load, and become broken.

Here is a full log from the GHA `pulsar --test spec` command:

<details>
<summary>GHA Test Log</summary>

```
/usr/bin/sudo apt-get install -y xvfb x11-xserver-utils
Reading package lists...
Building dependency tree...
Reading state information...
xvfb is already the newest version (2:1.20.13-1ubuntu1~20.04.12).
Suggested packages:
  nickle cairo-[5](https://github.com/pulsar-cooperative/linter-shellcheck-pulsar/actions/runs/7366657853/job/20049125911?pr=1#step:9:6)c xorg-docs-core
The following NEW packages will be installed:
  x11-xserver-utils
0 upgraded, 1 newly installed, 0 to remove and 14 not upgraded.
Need to get 1[6](https://github.com/pulsar-cooperative/linter-shellcheck-pulsar/actions/runs/7366657853/job/20049125911?pr=1#step:9:7)2 kB of archives.
After this operation, 51[7](https://github.com/pulsar-cooperative/linter-shellcheck-pulsar/actions/runs/7366657853/job/20049125911?pr=1#step:9:8) kB of additional disk space will be used.
Get:1 file:/etc/apt/apt-mirrors.txt Mirrorlist [142 B]
Get:2 http://azure.archive.ubuntu.com/ubuntu focal/main amd64 x11-xserver-utils amd64 7.7+[8](https://github.com/pulsar-cooperative/linter-shellcheck-pulsar/actions/runs/7366657853/job/20049125911?pr=1#step:9:9) [162 kB]
Fetched 162 kB in 0s (2606 kB/s)
Selecting previously unselected package x11-xserver-utils.
(Reading database ... 
(Reading database ... 5%
(Reading database ... 10%
(Reading database ... 15%
(Reading database ... 20%
(Reading database ... 25%
(Reading database ... 30%
(Reading database ... 35%
(Reading database ... 40%
(Reading database ... 45%
(Reading database ... 50%
(Reading database ... 55%
(Reading database ... 60%
(Reading database ... 65%
(Reading database ... 70%
(Reading database ... 75%
(Reading database ... 80%
(Reading database ... 85%
(Reading database ... [9](https://github.com/pulsar-cooperative/linter-shellcheck-pulsar/actions/runs/7366657853/job/20049125911?pr=1#step:9:10)0%
(Reading database ... 95%
(Reading database ... [10](https://github.com/pulsar-cooperative/linter-shellcheck-pulsar/actions/runs/7366657853/job/20049125911?pr=1#step:9:11)0%
(Reading database ... 300930 files and directories currently installed.)
Preparing to unpack .../x[11](https://github.com/pulsar-cooperative/linter-shellcheck-pulsar/actions/runs/7366657853/job/20049125911?pr=1#step:9:12)-xserver-utils_7.7+8_amd64.deb ...
Unpacking x11-xserver-utils (7.7+8) ...
Setting up x11-xserver-utils (7.7+8) ...
Processing triggers for man-db (2.9.1-1) ...
Command: pulsar --test spec
/usr/bin/xvfb-run --auto-servernum pulsar --test spec
(node:2611) Electron: Loading non-context-aware native module in renderer: '/tmp/.org.chromium.Chromium.73SJBF'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(Use `exe --trace-warnings ...` to show where the warning was created)
(node:2611) Electron: Loading non-context-aware native module in renderer: '/tmp/.org.chromium.Chromium.IVXvRD'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:2611) Electron: Loading non-context-aware native module in renderer: '/tmp/.org.chromium.Chromium.8rF8uE'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:2611) Electron: Loading non-context-aware native module in renderer: '/tmp/.org.chromium.Chromium.99m1XD'. This is deprecated, see https://github.com/electron/electron/issues/18397.
(node:2611) [DEP0005] DeprecationWarning: Buffer() is deprecated due to security and usability issues. Please use the Buffer.alloc(), Buffer.allocUnsafe(), or Buffer.from() methods instead.
(node:2611) Electron: Loading non-context-aware native module in renderer: '/opt/Pulsar/resources/app.asar.unpacked/node_modules/tree-sitter-bash/build/Release/tree_sitter_bash_binding.node'. This is deprecated, see https://github.com/electron/electron/issues/18397.
FFFFFF

The ShellCheck provider for Linter
  implements useProjectCwd and
    it uses file-relative source= directives by default
      Error: Failed to load package 'linter-shellcheck'
        at PackageManager.activatePackage (/opt/Pulsar/resources/app.asar/src/package-manager.js:776:29)
        at linter-shellcheck-spec.js:22:45
        at jasmine.Spec.middleware (/home/runner/work/linter-shellcheck-pulsar/linter-shellcheck-pulsar/node_modules/jasmine-fix/index.js:49:21)
      AssertionError: filePath must be a string
        AssertionError [ERR_ASSERTION] [ERR_ASSERTION]: filePath must be a string
        at validate (/home/runner/work/linter-shellcheck-pulsar/linter-shellcheck-pulsar/node_modules/sb-exec/lib/helpers.js:80:24)
        at execProxy (/home/runner/work/linter-shellcheck-pulsar/linter-shellcheck-pulsar/node_modules/sb-exec/lib/index.js:152:39)
        at Object.exec (/home/runner/work/linter-shellcheck-pulsar/linter-shellcheck-pulsar/node_modules/atom-linter/lib/helpers.js:105:19)
        at lint (main.js:79:24)
        at linter-shellcheck-spec.js:61:30
    it errors for file-relative source= path with useProjectCwd = true
      Error: Failed to load package 'linter-shellcheck'
        at PackageManager.activatePackage (/opt/Pulsar/resources/app.asar/src/package-manager.js:776:29)
        at linter-shellcheck-spec.js:22:45
        at jasmine.Spec.middleware (/home/runner/work/linter-shellcheck-pulsar/linter-shellcheck-pulsar/node_modules/jasmine-fix/index.js:49:21)
      AssertionError: filePath must be a string
        AssertionError [ERR_ASSERTION] [ERR_ASSERTION]: filePath must be a string
        at validate (/home/runner/work/linter-shellcheck-pulsar/linter-shellcheck-pulsar/node_modules/sb-exec/lib/helpers.js:80:24)
        at execProxy (/home/runner/work/linter-shellcheck-pulsar/linter-shellcheck-pulsar/node_modules/sb-exec/lib/index.js:152:39)
        at Object.exec (/home/runner/work/linter-shellcheck-pulsar/linter-shellcheck-pulsar/node_modules/atom-linter/lib/helpers.js:105:19)
        at lint (main.js:79:24)
        at linter-shellcheck-spec.js:68:30
    it uses project-relative source= directives via setting (based at fixtures/)
      Error: Failed to load package 'linter-shellcheck'
        at PackageManager.activatePackage (/opt/Pulsar/resources/app.asar/src/package-manager.js:776:29)
        at linter-shellcheck-spec.js:22:45
        at jasmine.Spec.middleware (/home/runner/work/linter-shellcheck-pulsar/linter-shellcheck-pulsar/node_modules/jasmine-fix/index.js:49:21)
      AssertionError: filePath must be a string
        AssertionError [ERR_ASSERTION] [ERR_ASSERTION]: filePath must be a string
        at validate (/home/runner/work/linter-shellcheck-pulsar/linter-shellcheck-pulsar/node_modules/sb-exec/lib/helpers.js:80:24)
        at execProxy (/home/runner/work/linter-shellcheck-pulsar/linter-shellcheck-pulsar/node_modules/sb-exec/lib/index.js:152:39)
        at Object.exec (/home/runner/work/linter-shellcheck-pulsar/linter-shellcheck-pulsar/node_modules/atom-linter/lib/helpers.js:105:19)
        at lint (main.js:79:24)
        at linter-shellcheck-spec.js:76:30
    it errors for project-relative source= path with useProjectCwd = false (based at fixtures/)
      Error: Failed to load package 'linter-shellcheck'
        at PackageManager.activatePackage (/opt/Pulsar/resources/app.asar/src/package-manager.js:776:29)
        at linter-shellcheck-spec.js:22:45
        at jasmine.Spec.middleware (/home/runner/work/linter-shellcheck-pulsar/linter-shellcheck-pulsar/node_modules/jasmine-fix/index.js:49:21)
      AssertionError: filePath must be a string
        AssertionError [ERR_ASSERTION] [ERR_ASSERTION]: filePath must be a string
        at validate (/home/runner/work/linter-shellcheck-pulsar/linter-shellcheck-pulsar/node_modules/sb-exec/lib/helpers.js:80:24)
        at execProxy (/home/runner/work/linter-shellcheck-pulsar/linter-shellcheck-pulsar/node_modules/sb-exec/lib/index.js:152:39)
        at Object.exec (/home/runner/work/linter-shellcheck-pulsar/linter-shellcheck-pulsar/node_modules/atom-linter/lib/helpers.js:105:19)
        at lint (main.js:79:24)
        at linter-shellcheck-spec.js:83:30
  it finds nothing wrong with a valid file
    Error: Failed to load package 'linter-shellcheck'
      at PackageManager.activatePackage (/opt/Pulsar/resources/app.asar/src/package-manager.js:776:29)
      at linter-shellcheck-spec.js:22:45
      at jasmine.Spec.middleware (/home/runner/work/linter-shellcheck-pulsar/linter-shellcheck-pulsar/node_modules/jasmine-fix/index.js:49:21)
    AssertionError: filePath must be a string
      AssertionError [ERR_ASSERTION] [ERR_ASSERTION]: filePath must be a string
      at validate (/home/runner/work/linter-shellcheck-pulsar/linter-shellcheck-pulsar/node_modules/sb-exec/lib/helpers.js:80:24)
      at execProxy (/home/runner/work/linter-shellcheck-pulsar/linter-shellcheck-pulsar/node_modules/sb-exec/lib/index.js:152:39)
      at Object.exec (/home/runner/work/linter-shellcheck-pulsar/linter-shellcheck-pulsar/node_modules/atom-linter/lib/helpers.js:105:19)
      at lint (main.js:79:24)
      at linter-shellcheck-spec.js:33:28
  it handles messages from ShellCheck
    Error: Failed to load package 'linter-shellcheck'
      at PackageManager.activatePackage (/opt/Pulsar/resources/app.asar/src/package-manager.js:776:29)
      at linter-shellcheck-spec.js:22:45
      at jasmine.Spec.middleware (/home/runner/work/linter-shellcheck-pulsar/linter-shellcheck-pulsar/node_modules/jasmine-fix/index.js:49:21)
    AssertionError: filePath must be a string
      AssertionError [ERR_ASSERTION] [ERR_ASSERTION]: filePath must be a string
      at validate (/home/runner/work/linter-shellcheck-pulsar/linter-shellcheck-pulsar/node_modules/sb-exec/lib/helpers.js:80:24)
      at execProxy (/home/runner/work/linter-shellcheck-pulsar/linter-shellcheck-pulsar/node_modules/sb-exec/lib/index.js:152:39)
      at Object.exec (/home/runner/work/linter-shellcheck-pulsar/linter-shellcheck-pulsar/node_modules/atom-linter/lib/helpers.js:105:19)
      at lint (main.js:79:24)
      at linter-shellcheck-spec.js:42:28


Finished in 0.421 seconds
6 tests, [12](https://github.com/pulsar-cooperative/linter-shellcheck-pulsar/actions/runs/7366657853/job/20049125911?pr=1#step:9:13) assertions, 12 failures, 0 skipped

/usr/bin/bash /home/runner/work/_actions/coactions/setup-xvfb/v1.0.1/dist/cleanup.sh
/usr/bin/bash: /home/runner/work/_actions/coactions/setup-xvfb/v1.0.1/dist/cleanup.sh: No such file or directory
Error: The process '/usr/bin/xvfb-run' failed with exit code 1
```

</details>

--- a/lib/main.js
+++ b/lib/main.js
@@ -19,7 +19,7 @@ const createURL = (text) => {
 
 export default {
   activate() {
-    require('atom-package-deps').install('linter-shellcheck');
+    require('atom-package-deps').install('linter-shellcheck-pulsar');
 
     this.subscriptions = new CompositeDisposable();
     this.subscriptions.add(

--- a/lib/main.js
+++ b/lib/main.js
@@ -23,16 +23,16 @@ export default {
 
     this.subscriptions = new CompositeDisposable();
     this.subscriptions.add(
-      atom.config.observe('linter-shellcheck.shellcheckExecutablePath', (value) => {
+      atom.config.observe('linter-shellcheck-pulsar.shellcheckExecutablePath', (value) => {
         this.executablePath = value;
       }),
-      atom.config.observe('linter-shellcheck.enableNotice', (value) => {
+      atom.config.observe('linter-shellcheck-pulsar.enableNotice', (value) => {
         this.enableNotice = value;
       }),
-      atom.config.observe('linter-shellcheck.userParameters', (value) => {
+      atom.config.observe('linter-shellcheck-pulsar.userParameters', (value) => {
         this.userParameters = value.trim().split(' ').filter(Boolean);
       }),
-      atom.config.observe('linter-shellcheck.useProjectCwd', (value) => {
+      atom.config.observe('linter-shellcheck-pulsar.useProjectCwd', (value) => {
         this.useProjectCwd = value;
       }),
     );

--- a/spec/linter-shellcheck-spec.js
+++ b/spec/linter-shellcheck-spec.js
@@ -36,7 +36,7 @@ describe('The ShellCheck provider for Linter', () => {
   });
 
   it('handles messages from ShellCheck', async () => {
-    const expectedExcerpt = 'Tips depend on target shell and yours is unknown. Add a shebang. [SC2148]';
+    const expectedExcerpt = 'Tips depend on target shell and yours is unknown. Add a shebang or a `shell` directive. [SC2148]';
     const expectedURL = 'https://github.com/koalaman/shellcheck/wiki/SC2148';
     const editor = await atom.workspace.open(badPath);
     const messages = await lint(editor);

--- a/spec/linter-shellcheck-spec.js
+++ b/spec/linter-shellcheck-spec.js
@@ -36,14 +36,17 @@ describe('The ShellCheck provider for Linter', () => {
   });
 
   it('handles messages from ShellCheck', async () => {
-    const expectedExcerpt = "Tips depend on target shell and yours is unknown. Add a shebang or a 'shell' directive. [SC2148]";
+    const expectedExcerpt = [
+      "Tips depend on target shell and yours is unknown. Add a shebang. [SC2148]",
+      "Tips depend on target shell and yours is unknown. Add a shebang or a 'shell' directive. [SC2148]"
+    ];
     const expectedURL = 'https://github.com/koalaman/shellcheck/wiki/SC2148';
     const editor = await atom.workspace.open(badPath);
     const messages = await lint(editor);
 
     expect(messages.length).toBe(1);
     expect(messages[0].severity).toBe('error');
-    expect(messages[0].excerpt).toBe(expectedExcerpt);
+    expect(expectedExcerpt.includes(messages[0].excerpt)).toBe(true);
     expect(messages[0].url).toBe(expectedURL);
     expect(messages[0].location.file).toBe(badPath);
     expect(messages[0].location.position).toEqual([[0, 0], [0, 4]]);

--- a/spec/linter-shellcheck-spec.js
+++ b/spec/linter-shellcheck-spec.js
@@ -19,7 +19,7 @@ describe('The ShellCheck provider for Linter', () => {
 
     // Info about this beforeEach() implementation:
     // https://github.com/AtomLinter/Meta/issues/15
-    const activationPromise = atom.packages.activatePackage('linter-shellcheck');
+    const activationPromise = atom.packages.activatePackage('linter-shellcheck-pulsar');
 
     await atom.packages.activatePackage('language-shellscript');
     await atom.workspace.open(cleanPath);
@@ -51,19 +51,19 @@ describe('The ShellCheck provider for Linter', () => {
 
   describe('implements useProjectCwd and', () => {
     beforeEach(async () => {
-      atom.config.set('linter-shellcheck.userParameters', '-x');
-      atom.config.set('linter-shellcheck.enableNotice', true);
+      atom.config.set('linter-shellcheck-pulsar.userParameters', '-x');
+      atom.config.set('linter-shellcheck-pulsar.enableNotice', true);
     });
 
     it('uses file-relative source= directives by default', async () => {
-      atom.config.set('linter-shellcheck.useProjectCwd', false);
+      atom.config.set('linter-shellcheck-pulsar.useProjectCwd', false);
       const editor = await atom.workspace.open(sourceFileRelativePath);
       const messages = await lint(editor);
       expect(messages.length).toBe(0);
     });
 
     it('errors for file-relative source= path with useProjectCwd = true', async () => {
-      atom.config.set('linter-shellcheck.useProjectCwd', true);
+      atom.config.set('linter-shellcheck-pulsar.useProjectCwd', true);
       const editor = await atom.workspace.open(sourceFileRelativePath);
       const messages = await lint(editor);
       expect(messages.length).toBe(1);
@@ -71,14 +71,14 @@ describe('The ShellCheck provider for Linter', () => {
     });
 
     it('uses project-relative source= directives via setting (based at fixtures/)', async () => {
-      atom.config.set('linter-shellcheck.useProjectCwd', true);
+      atom.config.set('linter-shellcheck-pulsar.useProjectCwd', true);
       const editor = await atom.workspace.open(sourceProjectRelativePath);
       const messages = await lint(editor);
       expect(messages.length).toBe(0);
     });
 
     it('errors for project-relative source= path with useProjectCwd = false (based at fixtures/)', async () => {
-      atom.config.set('linter-shellcheck.useProjectCwd', false);
+      atom.config.set('linter-shellcheck-pulsar.useProjectCwd', false);
       const editor = await atom.workspace.open(sourceProjectRelativePath);
       const messages = await lint(editor);
       expect(messages.length).toBe(1);

--- a/spec/linter-shellcheck-spec.js
+++ b/spec/linter-shellcheck-spec.js
@@ -36,7 +36,7 @@ describe('The ShellCheck provider for Linter', () => {
   });
 
   it('handles messages from ShellCheck', async () => {
-    const expectedExcerpt = 'Tips depend on target shell and yours is unknown. Add a shebang or a `shell` directive. [SC2148]';
+    const expectedExcerpt = "Tips depend on target shell and yours is unknown. Add a shebang or a 'shell' directive. [SC2148]";
     const expectedURL = 'https://github.com/koalaman/shellcheck/wiki/SC2148';
     const editor = await atom.workspace.open(badPath);
     const messages = await lint(editor);

